mautic/mautic
# [UX] Scheduling sending, availability and activation

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

This PR addresses users confusion re. status labels.

Now we have 11 variations.

**Active**:
- Active
- Inactive
- Not started yet
- Expired on schedule

**Available**:
- Available for use
- Unavailable
- Scheduled for upcoming use
- No longer available

**Sending**:
- Sending
- Unavailable
- Scheduled for future send
- Sending period expired

---

This PR also:
- Adds a tag to indicate the type of email, dynamic content, SMS and form
- Adds an icon on dynamic contents on their list page to indicate if they're campaign based
- Improves the translation re. variantes to add a mention to A/B as "variant" only could feel out of context for some people
- Adds a tag to indicate if a landing page is preference center
- Includes logic to enable or disable scheduling fields depending on the "Available for use" toggle
- Includes dynamic labels that change as you set scheduling options to tell the user how the item will behave


https://github.com/user-attachments/assets/fce62798-7165-465c-9b1a-a3272787bc30

![image](https://github.com/user-attachments/assets/8068b034-13c4-435e-a717-23e4d0fdcbdc)

![image](https://github.com/user-attachments/assets/7725c85b-3e3b-443c-b621-968dcfbd6f64)

![image](https://github.com/user-attachments/assets/8cb72dd2-dc3c-4e66-b5fd-750a34c2bd26)

![image](https://github.com/user-attachments/assets/99eacdfd-450d-4069-83b9-721605632e20)
![image](https://github.com/user-attachments/assets/c0963789-504f-4d2a-860b-2af8b2b2605d)
![image](https://github.com/user-attachments/assets/e36bf4e9-3855-4279-8531-038a96525d0c)
![image](https://github.com/user-attachments/assets/3c04c420-7041-47f4-9c70-f7ae77958800)



## Logic

### Availability (asset as example)
When **Available for use** is **Yes**:
- The resource becomes available and can be downloaded accessing the link, including the green Available tag in its details page
- If you set an Available from date, it becomes "Not started yet" and contacts cannot download accessing the link, it'll show a blue tag
- When the Available from date comes, it becomes "Available for use" with green tag and contacts can download accessing the link again
- If you set an Unavailable from date, just defining it does nothing on the status until it comes
- When the date for Unavailable from comes, it becomes "Expired on schedule" with blue tag and the respective status message, and the download link cannot be accessed anymore

When **Available for use** is **No**:
- Contacts cannot download the asset accessing the link
- No matter what you set for Available from or Unavailable from, none has effect
- The Available from field doesn't make the asset become available and downloadable
- It'll be always "Unavailable" label

### Sending (segment email as example)
When **Available for use** is **Yes**:
- The email becomes available and can be sent, including the green Available tag in its details page (only if Start sending at is empty)
- If you set a Start sending at date, it becomes pending with the blue label "Scheduled for future send" to indicate the system is waiting for the date to come so it can start sending
- When the Start sending at date comes, it becomes "Sending" with green tag and the system starts sending emails in batches to the contacts
- If you set a Stop sending at date for the email without a Start sending at date, just defining it does nothing on the status until it comes (before the Stop sending at date it'll keep Available for use but won't send; after the Stop sending at date, it'll turn into Unavailable)
- When the date for Stop sending at comes and you defined a Start sending at date previously, it becomes expired with the blue label "Sending period expired" and stops sending emails that started on the start date

When **Available for use** is **No**:
- System users can't send emails
- No matter what you set for Start sending at date or Stop sending at, none has effect
- Scheduled (using Start and Stop sending at) emails are not sent
- The Start sending at field doesn't make the email become available
- It'll be always "Unavailable" label







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Emails > New > Select the Segment type, play with the scheduling options then Save & Close and see the tags
3. Now create a Triggered email repeating the same process
4. Create A/B variations of these emails and see tags
5. Clone these emails to set as a different translation from the first ones and check tags
6. Repeat the process for SMS, landing pages, forms
7. Open Dynamic Content > New > Select an option for "Is campaign based" then Save & Close and see the tags
8. Create another Dynamic Content, but selecting the opposite for "Is campaign based" then Save & Close and see the tags
